internal/backends/linuxkms/calloop_backend.rs

```diff
@@ -240,9 +238,16 @@ impl i_slint_core::platform::Platform for Backend {
         while !quit_loop.load(std::sync::atomic::Ordering::Acquire) {
             i_slint_core::platform::update_timers_and_animations();
 
-            adapter.render_if_needed(mouse_position_property.as_ref())?;
+            if let Some(adapter) = self.window.borrow().as_ref() {
+                adapter.render_if_needed(mouse_position_property.as_ref())?;
+            }
 
-            let next_timeout = if adapter.window().has_active_animations() {
+            let next_timeout = if self
+                .window
+                .borrow()
+                .as_ref()
+                .map_or(false, |adapter| adapter.window().has_active_animations())
+            {
                 Some(std::time::Duration::from_millis(16))
```

Comment:
Not part of this patch, but i think that's not correct.
 - what if the next timer is before 16ms?
 - This is 16ms after the frame was rendered.  So say that rendering a frame takes a bit under 16ms,  we could still render at 60fps,  but with this wait, we would max at 30fps
 
 Ideally we should throttle based on the screen or something. Not sure how.
 But worst case, we should take 16ms since the end of the previous frame, not this frame.

Reply:
Yes, it's not correct. If a timer expires before 16ms we should indeed use that as timeout. And regarding the 16ms when there's no timer: This is indeed also not correct. I have a TODO to merge this with the page flip event:

If we have active animations, we should wait until we flipped pages, but in the meantime we should still spin the event loop to react to user input. That would probably also help with those spurious lib input messages about slow event processing. (Right now we wait blockingly for page flip, during which we don't process input)